### PR TITLE
Create a package for the config mock.

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

This creates a dedicated package for the config mock. This will allow us to reuse this mock for the config component and outside project.

This PR is better reviewed commit by commit. The 2nd commit was done using `gopatch` tool and should not need reviewing.

### Describe how to test/QA your changes

Unit tests are enough to validate this change.

### CODE REVIEW

**Review by guyarb — APPROVED** [review]
LGTM for usm

File: pkg/config/mock/mock.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+// SetKnown is used for setting configuration in tests
+func (c *MockConfig) SetKnown(key string) {
+	c.Config.SetKnown(key)
+}
+
+// Mock is creating and returning a mock config
+func New(t testing.TB) *MockConfig {
+	// We only check isConfigMocked when registering a cleanup function. 'isConfigMocked' avoids nested calls to
+	// Mock to reset the config to a blank state. This way we have only one mock per test and test helpers can call
+	// Mock.
```
Comment (ogaca-dd) [inline pkg/config/mock/mock.go:46]: Do you have more context about this comment?
In unit tests, you mostly want to have a refresh instance each time you call `New`?

**Review by ogaca-dd — APPROVED** [review]

**Review by davidor — APPROVED** [review]
:+1: for container-platform files

File: pkg/config/mock/mock.go (hunk 1 above)
Comment (hush-hush) [reply pkg/config/mock/mock.go:46]: This code is just moved from `pkg/config/mock.go` [here](https://github.com/DataDog/datadog-agent/pull/27482/files/0f3054ac10a29efd985a741467c0ddd15fc83255#diff-39d7ce767a00785ac9ed9c756fb0659948cbf220cbe33dc1e186dcaf6ea7ed01L44). This is used throughout the code base so test helpers can mock the config even if it was mocked before by the caller.

**Review by KevinFairise2 — APPROVED** [review]
Ok for the file owned by devx-loops

**Review by FlorentClarret — APPROVED** [review]

**Review by TCheruy — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

**Review by pducolin — APPROVED** [review]
Ok for files owned by @DataDog/agent-devx-loops

**Review by vickenty — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]
looks good for agent-apm

**Review by DylanLovesCoffee — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]
good from apm-onboarding

**Review by iglendd — APPROVED** [review]

**Review by ameske — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by daniel-taf — APPROVED** [review]

**Review by wiyu — APPROVED** [review]
👍 from processes

**Review by AliDatadog — APPROVED** [review]

**Pull request merged**